numpy/core/tests/test_mem_policy.py

```diff
@@ -0,0 +1,272 @@
+import asyncio
+import pytest
+import numpy as np
+import threading
+from numpy.testing import extbuild
+import sys
+
+
+@pytest.fixture
+def get_module(tmp_path):
+    """ Add a memory policy that returns a false pointer 64 bytes into the
+    actual allocation, and fill the prefix with some text. Then check at each
+    memory manipulation that the prefix exists, to make sure all alloc/realloc/
+    free/calloc go via the functions here.
+    """
+    if sys.platform.startswith('cygwin'):
+        pytest.skip('link fails on cygwin')
+    functions = [
+        ("set_secret_data_policy", "METH_NOARGS", """
+             const PyDataMem_Handler *old =
+                 PyDataMem_SetHandler(&secret_data_handler);
+             return PyCapsule_New((void *) old, NULL, NULL);
+         """),
+        ("set_old_policy", "METH_O", """
+             PyDataMem_Handler *old = NULL;
+             if (args != NULL && PyCapsule_CheckExact(args)) {
+                 old = (PyDataMem_Handler *) PyCapsule_GetPointer(args, NULL);
+             }
+             PyDataMem_SetHandler(old);
+             Py_RETURN_NONE;
+         """),
+    ]
+    prologue = '''
+        #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
+        #include <numpy/arrayobject.h>
+        /*
+         * This struct allows the dynamic configuration of the allocator funcs
+         * of the `secret_data_allocator`. It is provided here for
+         * demonstration purposes, as a valid `ctx` use-case scenario.
+         */
+        typedef struct {
+            void *(*malloc)(size_t);
+            void *(*calloc)(size_t, size_t);
+            void *(*realloc)(void *, size_t);
+            void (*free)(void *);
+        } SecretDataAllocatorFuncs;
+        NPY_NO_EXPORT void *
```

Comment:
```suggestion

        NPY_NO_EXPORT void *
```
A few newlines would help with readability, but just nitpicking.  I am happy especially as it is a test.